Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -352,8 +352,10 @@ WarpXParticleContainer::DepositCurrent (WarpXParIter& pti,
     // Jx, Jy and Jz have the same number of guard cells, hence it is sufficient to check for Jx
     const amrex::IntVect range = jx->nGrowVect() - shape_extent;
 #endif
-
-    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+#if !(defined(AMREX_DEBUG) || defined(AMREX_USE_ASSERTION))
+    amrex::ignore_unused(range);
+#endif
```

Comment:
Minimally more aesthetically pleasing:
```suggestion
    amrex::ignore_unused(range); // for release builds
```